sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java

```diff
@@ -253,7 +255,7 @@ public BlobClientBuilder endpoint(String endpoint) {
                 this.sharedKeyCredential = null;
             }
         } catch (MalformedURLException ex) {
-            throw new IllegalArgumentException("The Azure Storage Blob endpoint url is malformed.");
+            logger.logAndThrow(new IllegalArgumentException("The Azure Storage Blob endpoint url is malformed."));
```

Comment:
always end with 'return' statement after logger.logAndThrow()

Reply:
@mssfang can you work with @alzimmermsft and make sure this starts failing CI very early in Preview 3?  We need to quickly get out o the world that code can pass CI but break the build after merge.